pkg/utils/hwaddr/hwaddr.go

```diff
@@ -15,49 +15,27 @@
 package hwaddr
 
 import (
-	"fmt"
+	"math/rand"
 	"net"
 )
 
-const (
-	ipRelevantByteLen      = 4
-	PrivateMACPrefixString = "0a:58"
-)
-
-var (
-	// private mac prefix safe to use
-	PrivateMACPrefix = []byte{0x0a, 0x58}
-)
-
-type SupportIp4OnlyErr struct{ msg string }
-
-func (e SupportIp4OnlyErr) Error() string { return e.msg }
-
-type MacParseErr struct{ msg string }
-
-func (e MacParseErr) Error() string { return e.msg }
-
-type InvalidPrefixLengthErr struct{ msg string }
-
-func (e InvalidPrefixLengthErr) Error() string { return e.msg }
-
-// GenerateHardwareAddr4 generates 48 bit virtual mac addresses based on the IP4 input.
-func GenerateHardwareAddr4(ip net.IP, prefix []byte) (net.HardwareAddr, error) {
-	switch {
-
-	case ip.To4() == nil:
-		return nil, SupportIp4OnlyErr{msg: "GenerateHardwareAddr4 only supports valid IPv4 address as input"}
-
-	case len(prefix) != len(PrivateMACPrefix):
-		return nil, InvalidPrefixLengthErr{msg: fmt.Sprintf(
-			"Prefix has length %d instead  of %d", len(prefix), len(PrivateMACPrefix)),
-		}
-	}
-
-	ipByteLen := len(ip)
-	return (net.HardwareAddr)(
-		append(
-			prefix,
-			ip[ipByteLen-ipRelevantByteLen:ipByteLen]...),
-	), nil
+// The first 24 bits of the MAC represent the Organizationally Unique Identifier (OUI),
+// and the first byte of the MAC address has two special bits.
+// 1. The least-significant bit: 0 for unicast and 1 for multicast
+// 2. The second-least-significant bit: 0 for globally unique and 1 for locally administered
+// Since MAC address for container interfaces is locally administered and unicast, so we can
+// fix the two LSb of MSB to 10 and use any values for other bits. This will avoid any conflicts
```

Comment:
Minor typo: LSB

Reply:
emmm, actually LSb here means least-significant bit, not byte @jellonek

Reply:
Ough. Right!